File: frontend/public/sdk/termpilot.js
/* TermPilot read-only SDK. No mail send. No calendar write. */
(function (root) {
  function TermPilot(base) {
    this.base = (base || "http://127.0.0.1:8000").replace(/\/$/, "");
  }
  TermPilot.prototype.get = async function (path) {
    const r = await fetch(this.base + path);
    if (!r.ok) throw new Error(String(r.status));
    return r.json();
  };
  TermPilot.prototype.health = function () { return this.get("/health"); };
  TermPilot.prototype.tower = function () { return this.get("/tower"); };
  TermPilot.prototype.catalog = function () { return this.get("/llm/catalog"); };
  root.TermPilot = TermPilot;
})(typeof window !== "undefined" ? window : globalThis);

File: sdk/termpilot.py
"""TermPilot offline/read-only SDK. Does not send mail or write calendars."""

from __future__ import annotations

import json
import urllib.request


class TermPilot:
    def __init__(self, base: str = "http://127.0.0.1:8000") -> None:
        self.base = base.rstrip("/")

    def _get(self, path: str) -> object:
        with urllib.request.urlopen(self.base + path) as response:
            return json.load(response)

    def health(self) -> object:
        return self._get("/health")

    def tower(self) -> object:
        return self._get("/tower")

    def catalog(self) -> object:
        return self._get("/llm/catalog")
